src/lazy/system_reader.rs

```diff
@@ -228,11 +275,61 @@ impl<Encoding: Decoder, Input: IonInput> SystemReader<Encoding, Input> {
                 }
                 // Any other symbol is ignored
             }
-            // TODO: Implement shared symbol table imports
-            ExpandedValueRef::List(_) => {
-                return IonResult::decoding_error(
-                    "This implementation does not yet support shared symbol table imports",
-                );
+            ExpandedValueRef::List(list) => {
+                for value in list.iter() {
+                    let ExpandedValueRef::Struct(import) = value?.read()? else {
+                        // If there's a value in the imports list that isn't a struct, it's malformed.
+                        // Ignore that value.
+                        continue;
+                    };
+                    let name = match import.get("name")? {
+                        // If `name` is missing, a non-string, or the empty string, ignore this import.
+                        Some(ExpandedValueRef::String(s)) if !s.is_empty() => s,
+                        _ => continue,
+                    };
+                    let version: usize = match import.get("version")? {
+                        Some(ExpandedValueRef::Int(i)) if i > Int::ZERO => usize::try_from(i)
+                            .map_err(|_|
+                            IonError::decoding_error(format!("found a symbol table import (name='{name}') with a version number too high to support: {i}")),
+                        ),
+                        // If there's no version, a non-int version, or a version <= 0, we treat it
+                        // as version 1.
+                        _ => Ok(1),
+                    }?;
+
+                    let shared_table = match catalog.get_table_with_version(name.as_ref(), version) {
+                        Some(table) => table,
+                        None => return IonResult::decoding_error(
+                            format!("symbol table import failed, could not find table with name='{name}' and version={version}")
```

Comment:
🗺️ `ion-rust` does not implement fallback to older versions of an imported symbol table, choosing instead to treat the unavailability of a specific imported SST version as a hard error.

Ion implementations that offer this capability (e.g. `ion-java`) are forced to track three possible resolved symbol kinds:
1. A symbol with known text
2. A symbol that is known to have no text (`$0`)
3. A symbol that might have text but we don't have the context necessary to look it up

State 3 is defined to allow a reader to "soldier on" in the absence of the necessary symbol definitions, reading the parts of the file it _does_ have. Early versions of `ion-rust` preserved state 3, but it introduced considerable API complexity. We tabled support for it until a concrete use case surfaced.